Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/Wrappers/SpeckleBlockInstanceWrapper.cs

```diff
@@ -51,7 +52,20 @@ public InstanceProxy InstanceProxy
       UpdateTransformFromProxy();
     }
   }
-  public SpeckleBlockDefinitionWrapper? Definition { get; set; }
+
+  public SpeckleBlockDefinitionWrapper? Definition
+  {
+    get => _definition;
+    set
+    {
+      _definition = value;
+      if (_definition != null)
+      {
+        _definition.ApplicationId ??= Guid.NewGuid().ToString();
```

Comment:
this definition id should probably not be null at this point? I would actually throw here if it is null, instead of mutating the definition in the setter

Reply:
done -> [fix: throw if id null](https://github.com/specklesystems/speckle-sharp-connectors/pull/949/commits/2e2c8e173233670c1938415b3736958321cafb83)